src/IO/ReadHelpers.h

```diff
@@ -487,8 +488,66 @@ struct NullSink
     void push_back(char) {}
 };
 
-void parseUUID(const UInt8 * src36, UInt8 * dst16);
-void parseUUID(const UInt8 * src36, std::reverse_iterator<UInt8 *> dst16);
+template <typename IteratorSrc, typename IteratorDst>
+void parseHex(IteratorSrc src, IteratorDst dst, const size_t num_bytes)
+{
+    size_t src_pos = 0;
+    size_t dst_pos = 0;
+    for (; dst_pos < num_bytes; ++dst_pos)
+    {
+        dst[dst_pos] = UInt8(unhex(src[src_pos])) * 16 + UInt8(unhex(src[src_pos + 1]));
+        src_pos += 2;
+    }
+}
+
+template <bool with_separator>
+void parseUUID(const UInt8 * src36, UInt8 * dst16)
```

Comment:
This template is always used with explicit parameter. And it is implemented with two separate branches...

So, we can simply make two functions:

parseUUID
parseUUIDWithoutSeparator